contrib/99designs/gqlgen/tracer_test.go

```diff
@@ -0,0 +1,150 @@
+		if span.Tag(tag) == val {
+			return span
+		}
+	}
+	return nil
+}
+
+func TestRootSpan(t *testing.T) {
+	for name, tt := range map[string]struct {
+		clientOpts []client.Option
+		tracerOpts []Option
+		test       func(assert *assert.Assertions, spans []mocktracer.Span)
+	}{
+		"default": {
+			test: func(assert *assert.Assertions, spans []mocktracer.Span) {
+				root := findRootSpan(spans)
+				assert.NotNil(root)
+				assert.Equal(root.Tag(ext.ResourceName), defaultResourceName)
+				assert.Equal(root.Tag(ext.ServiceName), defaultServiceName)
+				assert.Equal(root.Tag(ext.SpanType), ext.SpanTypeGraphql)
+				assert.Nil(root.Tag(ext.EventSampleRate))
+			},
+		},
+		"OperationName": {
+			clientOpts: []client.Option{client.Operation("CreateTodo")},
+			test: func(assert *assert.Assertions, spans []mocktracer.Span) {
+				root := findRootSpan(spans)
+				assert.NotNil(root)
+				assert.Equal(root.Tag(ext.ResourceName), "CreateTodo")
+			},
+		},
+		"ServiceName": {
+			tracerOpts: []Option{WithServiceName("TodoServer")},
+			test: func(assert *assert.Assertions, spans []mocktracer.Span) {
+				root := findRootSpan(spans)
+				assert.NotNil(root)
+				assert.Equal(root.Tag(ext.ServiceName), "TodoServer")
+			},
+		},
+		"WithAnalytics/true": {
+			tracerOpts: []Option{WithAnalytics(true)},
+			test: func(assert *assert.Assertions, spans []mocktracer.Span) {
+				root := findRootSpan(spans)
+				assert.NotNil(root)
+				assert.Equal(1.0, root.Tag(ext.EventSampleRate))
+			},
+		},
+		"WithAnalytics/false": {
+			tracerOpts: []Option{WithAnalytics(false)},
+			test: func(assert *assert.Assertions, spans []mocktracer.Span) {
+				root := findRootSpan(spans)
+				assert.NotNil(root)
+				assert.Nil(root.Tag(ext.EventSampleRate))
+			},
+		},
+		"WithAnalyticsRate": {
+			tracerOpts: []Option{WithAnalyticsRate(0.5)},
+			test: func(assert *assert.Assertions, spans []mocktracer.Span) {
+				root := findRootSpan(spans)
+				assert.NotNil(root)
+				assert.Equal(0.5, root.Tag(ext.EventSampleRate))
+			},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+			mt := mocktracer.Start()
+			defer mt.Stop()
+			c := newTodoClient(New(tt.tracerOpts...))
+
+			var createResp struct {
+				CreateTodo struct{ ID string }
+			}
+			err := c.Post(`mutation CreateTodo{ createTodo(todo: {text: "todo text"}) {id} }`, &createResp, tt.clientOpts...)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			spans := mt.FinishedSpans()
+			tt.test(assert, spans)
+		})
+	}
+}
+
+func TestResolver(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+	c := newTodoClient(New())
+
+	var createResp struct {
+		CreateTodo struct{ ID string }
+	}
+	err := c.Post(`mutation CreateTodo{ createTodo(todo: {text: "todo text"}) {id} }`, &createResp)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	spans := mt.FinishedSpans()
+	span := findSpanWithTag(spans, ext.ResourceName, "Field_createTodo")
```

Comment:
Can we remove this function and check all spans created? We want to make sure that all of them work as expected.